### PR TITLE
SIDE-1652 Add disambiguation whisper to C#

### PR DESCRIPTION
**Pull request by BrettMugglinOlive**
Shoulda been there

### CODE REVIEW

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Disambiguation.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    /// Base class for Whisper Disambiguation responses.
+    /// </summary>
+    public interface IWhisperDisambiguationResponse
+    {
+    }
+
+    /// <summary>
+    /// Represents a selection of an item
+    /// </summary>
+    public struct DisambiguationResponse : IWhisperDisambiguationResponse
```
Comment (richseviora) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Disambiguation.cs:13]: I'm thinking that whenever the namespace includes the relevant noun that it's perhaps redundant to include the noun again in the object name.

So this becomes `OliveHelpsLDK.Whispers.Disambiguation.Response`. 

@RusGraf what do you think?

Comment (richseviora) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Disambiguation.cs:13]: And it's entirely possible I have not been consistent/following this myself :)

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/WhisperClient.cs [hunk 2]
```diff
@@ -67,6 +68,27 @@ public Task<bool> ConfirmAsync(WhisperConfirm message, CancellationToken cancell
...
+        {
+            var request = Builder.BuildRequest(message, CreateSession());
+            var call = Client.WhisperDisambiguation(request, new CallOptions(cancellationToken: cancellationToken));
+            var loggedParser = LoggedParser<Proto.WhisperDisambiguationStreamResponse, IWhisperDisambiguationResponse>(response =>
+                Parser.ParseResponse(response)); 
+            return new StreamingCall<WhisperDisambiguationStreamResponse, IWhisperDisambiguationResponse>(call, loggedParser);
+        }
+
+        /* 
+        Need to look into the parser
```
Comment (richseviora) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/WhisperClient.cs:81]: What needs to be checked?

**Review by richseviora — APPROVED** [review]
Overall fine - one opportunity to rename some classes but not sure if we should

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Disambiguation.cs (hunk 1 above)
Comment (RusGraf) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Disambiguation.cs:13]: I would think you are correct if the situation be like:
```
public class WhisperDisambiguation 
{
    public Response {get; set;} // not public WhisperDisambiguationResponse {get; set;}
}
```

But it is totally correct to have Foo : IFoo. To be honest I don't mind long class or struct names. As long as they describing the intent.

**Review by RusGraf — CHANGES_REQUESTED** [review]
I think there is an outstanding question about commented out code. Please respond. Otherwise all good.

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Disambiguation.cs (hunk 1 above)
Comment (BrettMugglinOlive) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/Disambiguation.cs:13]: Ok, I can update those names

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/WhisperClient.cs (hunk 2 above)
Comment (BrettMugglinOlive) [reply ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Whispers/WhisperClient.cs:81]: Nothing, that code block needs to be removed

**Review by RusGraf — APPROVED** [review]

**Pull request merged**